task_21/toma.lobanova/src/main/java/com/hillel/tasks/train/Train.java

```diff
@@ -0,0 +1,70 @@
+package com.hillel.tasks.train;
+
+import java.util.ArrayList;
+
+public class Train {
+  ArrayList<Coach> coaches;
+
+  public Train(int coachesNumber) {
+    coaches = new ArrayList<Coach>();
+    if (coachesNumber <= 0) {
+      throw new NonPositiveCoachesNumber("The number of coaches in the train should be a positive number");
+    }
+    for (int i = 0; i < coachesNumber; i++) {
+      coaches.add(new Coach());
+    }
+  }
+
+  public boolean isTrainFull() {
+    for (int i = 0; i < coaches.size(); i++) {
+      if (!coaches.get(i).isFull()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public void bookPlace(int coachNumber, int placeNumber) {
+    if (checkIfValidArguments(coachNumber, placeNumber)) {
```

Comment:
I would check for wrong conditions first and then the main logic, it would eliminate the nested if's and else's:
if (!checkIfValidArguments(...)) {
   throw new ....;
}
the rest of the code goes here without the need of "else" block